config/cmake/modules/FindSLEPc.cmake

```diff
@@ -0,0 +1,278 @@
+
+  # Add variables to CMake cache and mark as advanced
+  set(SLEPC_INCLUDE_DIRS ${SLEPC_INCLUDE_DIRS} CACHE STRING "SLEPc include paths." FORCE)
+  set(SLEPC_LIBRARIES ${SLEPC_LIBRARY} CACHE STRING "SLEPc libraries." FORCE)
+  mark_as_advanced(SLEPC_INCLUDE_DIRS SLEPC_LIBRARIES)
+endif()
+
+if (DOLFIN_SKIP_BUILD_TESTS)
+  set(SLEPC_TEST_RUNS TRUE)
+  set(SLEPC_VERSION "UNKNOWN")
+  set(SLEPC_VERSION_OK TRUE)
+elseif (SLEPC_LIBRARIES AND SLEPC_INCLUDE_DIRS)
+
+  # Set flags for building test program
+  set(CMAKE_REQUIRED_INCLUDES ${SLEPC_INCLUDE_DIRS} ${PETSC_INCLUDE_DIRS})
+  set(CMAKE_REQUIRED_LIBRARIES ${SLEPC_LIBRARIES} ${PETSC_LIBRARIES})
+
+  # Add MPI variables if MPI has been found
+  if (MPI_C_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES  ${CMAKE_REQUIRED_INCLUDES} ${MPI_C_INCLUDE_PATH})
+    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${MPI_C_LIBRARIES})
+    set(CMAKE_REQUIRED_FLAGS     "${CMAKE_REQUIRED_FLAGS} ${MPI_C_COMPILE_FLAGS}")
+  endif()
+
+  # Check SLEPc version
+  set(SLEPC_CONFIG_TEST_VERSION_CPP
+    "${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/slepc_config_test_version.cpp")
+  file(WRITE ${SLEPC_CONFIG_TEST_VERSION_CPP} "
+#include <iostream>
+#include \"slepcversion.h\"
+
+int main() {
+  std::cout << SLEPC_VERSION_MAJOR << \".\"
+	    << SLEPC_VERSION_MINOR << \".\"
+	    << SLEPC_VERSION_SUBMINOR;
+  return 0;
+}
+")
+
+  try_run(
+    SLEPC_CONFIG_TEST_VERSION_EXITCODE
+    SLEPC_CONFIG_TEST_VERSION_COMPILED
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${SLEPC_CONFIG_TEST_VERSION_CPP}
+    CMAKE_FLAGS
+      "-DINCLUDE_DIRECTORIES:STRING=${CMAKE_REQUIRED_INCLUDES}"
+    COMPILE_OUTPUT_VARIABLE COMPILE_OUTPUT
+    RUN_OUTPUT_VARIABLE OUTPUT
+    )
+
+  if (SLEPC_CONFIG_TEST_VERSION_EXITCODE EQUAL 0)
+    set(SLEPC_VERSION ${OUTPUT} CACHE STRING STRING)
+    string(REPLACE "." ";" SLEPC_VERSION_LIST ${SLEPC_VERSION})
+    list(GET SLEPC_VERSION_LIST 0 SLEPC_VERSION_MAJOR)
+    list(GET SLEPC_VERSION_LIST 1 SLEPC_VERSION_MINOR)
+    list(GET SLEPC_VERSION_LIST 2 SLEPC_VERSION_SUBMINOR)
+    mark_as_advanced(SLEPC_VERSION)
+    mark_as_advanced(SLEPC_VERSION_MAJOR, SLEPC_VERSION_MINOR, SLEPC_VERSION_SUBMINOR)
+  endif()
+
+  if (SLEPc_FIND_VERSION)
+    # Check if version found is >= required version
+    if (NOT "${SLEPC_VERSION}" VERSION_LESS "${SLEPc_FIND_VERSION}")
+      set(SLEPC_VERSION_OK TRUE)
+    endif()
+  else()
+    # No specific version requested
+    set(SLEPC_VERSION_OK TRUE)
+  endif()
+  mark_as_advanced(SLEPC_VERSION_OK)
+
+  # Run SLEPc test program
+  set(SLEPC_TEST_LIB_CPP
+    "${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/slepc_test_lib.cpp")
+  file(WRITE ${SLEPC_TEST_LIB_CPP} "
+#include \"petsc.h\"
+#include \"slepceps.h\"
+int main()
+{
+  PetscErrorCode ierr;
+  int argc = 0;
+  char** argv = NULL;
+  ierr = SlepcInitialize(&argc, &argv, PETSC_NULL, PETSC_NULL);
+  EPS eps;
+  ierr = EPSCreate(PETSC_COMM_SELF, &eps); CHKERRQ(ierr);
+  //ierr = EPSSetFromOptions(eps); CHKERRQ(ierr);
```

Comment:
why this commented line?

Reply:
Not sure, I will try to simplify this file.